pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -322,10 +322,13 @@ def sat_ready_rhel(request):
 
 
 @pytest.fixture(scope='module')
-def module_sat_ready_rhels(request):
+def module_sat_ready_rhels(request, module_target_sat):
     deploy_args = get_deploy_args(request)
-    with Broker(**deploy_args, host_class=Satellite, _count=2) as hosts:
-        yield hosts
+    if 'build_sanity' not in request.config.option.markexpr:
```

Comment:
@jyejare This is not a good approach. This change will cause unnecessary call to `module_target_sat` and extra checkout in normal automation run.